src/java.base/share/classes/sun/security/provider/SHA3.java

```diff
@@ -103,22 +106,33 @@ private void implCompress0(byte[] b, int ofs) {
         keccak();
     }
 
+     void finishAbsorb() {
+        int numOfPadding =
+                setPaddingBytes(suffix, buffer, (int)(bytesProcessed % blockSize));
+        if (numOfPadding < 1) {
+            throw new ProviderException("Incorrect pad size: " + numOfPadding);
+        }
+        implCompress(buffer, 0);
+    }
+
     /**
      * Return the digest. Subclasses do not need to reset() themselves,
      * DigestBase calls implReset() when necessary.
      */
     void implDigest(byte[] out, int ofs) {
         byte[] byteState = new byte[8];
```

Comment:
If I do that, the performance drops a little. So I rather keep it here.

Reply:
I see. In fact, what if `byteState` is an instance variable outside the method? If a previous squeeze has left some bytes there, there is no need to call `asLittleEndian.set` again next time they are needed. Does it help in any performance gain? especially if the caller only squeeze a few bytes each time?

Reply:
Also, a comment will be nice here. Otherwise, some people might still want to move it.

Reply:
I have considered the instance variable as well.  It might help the performance of the case you mention (many small squeezes), but I don't think that would be a typical use case. I am adding the comment.

Reply:
ML-KEM only squeezes 3 bytes each time, but you probably already optimized that by storing enough bytes for later uses.